### PR TITLE
[MNG-7443] IT for consistent logging of optional selectors

### PR DESCRIPTION
**Pull request by Giovds**

### CODE REVIEW

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7443ConsistencyOfOptionalProjectsAndProfilesTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        verifier.executeGoals( Arrays.asList( "clean", "verify" ) );
+
+        final List<String> logLines = verifier.loadFile( verifier.getBasedir(), verifier.getLogFileName(), false );
+
+        int projectSelectorMissingCounter = 0;
+        int profileSelectorMissingCounter = 0;
+
+        for ( String logLine : logLines )
+        {
+            if ( logLine.equals( "[INFO] The requested optional projects [:does-not-exist] do not exist." ) )
```
Comment (mthmulders) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7443ConsistencyOfOptionalProjectsAndProfilesTest.java:55]: I would try to match on the most relevant part of the log line, e.g. "requested optional project" and ":does not exist". The other parts are not so important. Same goes for line 60.

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7443ConsistencyOfOptionalProjectsAndProfilesTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        super( "[4.0.0-alpha-1,)" );
+    }
+
+    public void testConsistentLoggingOfOptionalProfilesAndProjects() throws IOException, VerificationException
+    {
+        File testDir = ResourceExtractor.simpleExtractResources( getClass(),
+                "/mng-7443-consistency-of-optional-profiles-and-projects" );
+
+        Verifier verifier = newVerifier( testDir.getAbsolutePath() );
+        verifier.addCliOption( "-pl ?:does-not-exist" );
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7443ConsistencyOfOptionalProjectsAndProfilesTest.java:43]: These are two options, hence the space.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7443ConsistencyOfOptionalProjectsAndProfilesTest.java (hunk 2 above)
Comment (mthmulders) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7443ConsistencyOfOptionalProjectsAndProfilesTest.java]: Maybe the same for the line below this one, which now reads `-P?does-not-exist-either`...

**Comment by michael-o** [comment]
Running this now, the IT needs some non-functional improvements I will perform

**Pull request closed without merging**